src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DelegateCtorSignature.cs

```diff
@@ -29,6 +29,11 @@ public class DelegateCtorSignature : Signature
             _targetMethod = targetMethod;
             _methodToken = methodToken;
             _signatureContext = signatureContext;
+
+            // Ensure types in signature are loadable and resolvable, otherwise we'll fail later while emitting the signature
+            CompilerTypeSystemContext context = (CompilerTypeSystemContext)delegateType.Context;
```

Comment:
Not: The token resolver in the signatureContext has a strongly typed CompilerTypeSystemContext in it. We could expose that and avoid the type casts (and have the C# compiler statically ensure everything has the right type here) .

(Also applies elsewhere)

Reply:
Ok. I'll wait for the crossgen2 run to complete to get a read on test quality before making nit changes